test/FunctionalTests/Microsoft.OData.Core.Tests/Json/JsonWriterBaseTests.cs

```diff
@@ -601,6 +602,451 @@ public void WriteMixedJsonWithLargeValues()
+                jsonWriter.WriteValue(largeStringData);
+                jsonWriter.WriteValue(largeBase64Data);
+                jsonWriter.EndArrayScope();
+
+                jsonWriter.EndObjectScope();
+
+                jsonWriter.Flush();
+                stream.Seek(0, SeekOrigin.Begin);
+                using (StreamReader reader = new StreamReader(stream, encoding: Encoding.UTF8))
+                {
+                    string rawOutput = reader.ReadToEnd();
+                    string normalizedOutput = NormalizeJsonText(rawOutput);
+                    string normalizedExpectedOutput = NormalizeJsonText(expectedJson);
+                    Assert.Equal(normalizedExpectedOutput, normalizedOutput);
+                }
+            }
+        }
+
+        [Fact]
+        public void CorrectlyStreamsLargeByteArrayToOutput()
+        {
+            byte[] input = GenerateByteArray(1024 * 1024);
+
+            using (MemoryStream stream = new MemoryStream())
+            {
+                IJsonWriter jsonWriter = CreateJsonWriter(stream, false, Encoding.UTF8);
+                var st = jsonWriter.StartStreamValueScope();
+                WriteByteArrayInChunks(st, input);
+                jsonWriter.EndStreamValueScope();
+                jsonWriter.Flush();
+
+                stream.Seek(0, SeekOrigin.Begin);
+
+                string expected = Convert.ToBase64String(input);
+
+                using (StreamReader reader = new StreamReader(stream, encoding: Encoding.UTF8))
+                {
+                    string rawOutput = reader.ReadToEnd();
+                    Assert.Equal($"\"{expected}\"", rawOutput);
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData("text/plain")]
+        [InlineData("text/html")]
+        public void CorrectlyStreamsLargeStringsToOutput(string contentType)
+        {
+            int inputLength = 1024 * 1024; // 1MB
+            string input = new string('a', inputLength);
+
+            using (MemoryStream stream = new MemoryStream())
+            {
+                IJsonWriter jsonWriter = CreateJsonWriter(stream, false, Encoding.UTF8);
+
+                var tw = jsonWriter.StartTextWriterValueScope(contentType);
+
+                WriteLargeStringInChunks(tw, input);
+
+                jsonWriter.EndTextWriterValueScope();
+                jsonWriter.Flush();
+
+                stream.Seek(0, SeekOrigin.Begin);
+
+                using (StreamReader reader = new StreamReader(stream, encoding: Encoding.UTF8))
+                {
+                    string rawOutput = reader.ReadToEnd();
+                    Assert.Equal($"\"{input}\"", rawOutput);
+                }
+            }
+        }
+
+        [Fact]
+        public void CorrectlyStreamsLargeStringsToOutput_ApplicationJson_ContentType()
+        {
+            int inputLength = 1024 * 1024; // 1MB
+            string input = new string('a', inputLength);

```

Comment:
I think we should also have special characters.

Reply:
added tests with special characters